src/main/java/com/blackducksoftware/integration/jira/config/controller/action/BlackDuckConfigActions.java

```diff
@@ -0,0 +1,451 @@
+        final Optional<Integer> blackDuckProxyPort = settings.getBlackDuckProxyPort();
+        final String blackDuckProxyUser = settings.getBlackDuckProxyUser();
+        final String encBlackDuckProxyPassword = settings.getBlackDuckProxyPassword();
+
+        final BlackDuckHttpClient restConnection;
+        try {
+            final BlackDuckServerConfigBuilder configBuilder = new BlackDuckServerConfigBuilder();
+            configBuilder.setUrl(blackDuckUrl);
+            configBuilder.setApiToken(blackDuckApiToken);
+            configBuilder.setTimeout(blackDuckTimeout.orElse(300));
+            configBuilder.setTrustCert(blackDuckTrustCert);
+            configBuilder.setProxyHost(blackDuckProxyHost);
+            blackDuckProxyPort.ifPresent(configBuilder::setProxyPort);
+            configBuilder.setProxyUsername(blackDuckProxyUser);
+            configBuilder.setProxyPassword(encBlackDuckProxyPassword);
+
+            final BlackDuckServerConfig serverConfig;
+            try {
+                serverConfig = configBuilder.build();
+            } catch (final IllegalStateException e) {
+                logger.error("Error in Black Duck server configuration: " + e.getMessage());
+                throw new ConfigurationException(JiraConfigErrorStrings.CHECK_BLACKDUCK_SERVER_CONFIGURATION);
+            }
+
+            restConnection = serverConfig.createBlackDuckHttpClient(logger);
+        } catch (final IllegalArgumentException e) {
+            throw new ConfigurationException(JiraConfigErrorStrings.CHECK_BLACKDUCK_SERVER_CONFIGURATION + " :: " + e.getMessage());
+        }
+        return restConnection;
+    }
+
+    private List<String> getBlackDuckProjects(final BlackDuckServicesFactory blackDuckServicesFactory) throws ConfigurationException {
+        final List<String> blackDuckProjects = new ArrayList<>();
+        blackDuckProjects.add(BlackDuckProjectMappings.MAP_ALL_PROJECTS);
+
+        final ProjectService projectRequestService = blackDuckServicesFactory.createProjectService();
+        final List<ProjectView> blackDuckProjectItems;
+        try {
+            blackDuckProjectItems = projectRequestService.getAllProjectMatches(null);
+        } catch (final IntegrationException e) {
+            throw new ConfigurationException(e.getMessage());
+        }
+
+        if (blackDuckProjectItems != null && !blackDuckProjectItems.isEmpty()) {
+            for (final ProjectView project : blackDuckProjectItems) {
+                final List<String> allowedMethods = project.getAllowedMethods();
+                if (allowedMethods != null && !allowedMethods.isEmpty() && allowedMethods.contains("GET") && allowedMethods.contains("PUT")) {
+                    blackDuckProjects.add(project.getName());
+                }
+            }
+        }
+        return blackDuckProjects;
+    }
+
+    private void setBlackDuckPolicyRules(final BlackDuckServicesFactory blackDuckServicesFactory, final BlackDuckJiraConfigSerializable config) {
```

Comment:
We might want to abstract a few parts of this method based on its size and complexity.